### PR TITLE
Add DAL for UserProfileService #3

### PR DESCRIPTION
**Pull request by vladzhorov**

### CODE REVIEW

File: ServiceFinder/ServiceFinder.DAL/DalServiceExtension.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            services.AddScoped<SoftDeleteInterceptor>();
+
+            services.AddDbContext<AppDbContext>((serviceProvider, options) =>
+            {
+                var dbOptions = serviceProvider.GetRequiredService<IOptions<DatabaseOptions>>().Value;
+
+                options.UseNpgsql(dbOptions.ConnectionString)
+                 .AddInterceptors(
+                    serviceProvider.GetRequiredService<UpdateAuditableInterceptor>(),
+                    serviceProvider.GetRequiredService<SoftDeleteInterceptor>());
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.DAL/DalServiceExtension.cs:26]: Better to use optionsBuilder.AddInterceptors() to register interceptors. You can create objects via constructor

File: ServiceFinder/ServiceFinder.DAL/Entites/ServiceEntity.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+namespace ServiceFinder.DAL.Entites
+{
+    public class ServiceEntity : ISoftDeleteEntity, IAuditableEntity
+    {
+        public Guid Id { get; set; }
+        public Guid UserProfileId { get; set; }
+        public Guid ServiceCategoryId { get; set; }
+        public string Title { get; set; }
+        public ServiceCategoryEntity Category { get; set; }
+        public string Description { get; set; }
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.DAL/Entites/ServiceEntity.cs:14]: warnings not fixed for the file

File: ServiceFinder/ServiceFinder.DAL/Entites/ServiceCategoryEntity.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+namespace ServiceFinder.DAL.Entites
+{
+    public class ServiceCategoryEntity : ISoftDeleteEntity, IAuditableEntity
+    {
+        public Guid Id { get; set; }
+        public string Name { get; set; }
+        public string Description { get; set; }
+        public DateTime CreatedAt { get; set; }
+        public DateTime UpdatedAt { get; set; }
+        public bool IsDeleted { get; set; }
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.DAL/Entites/ServiceCategoryEntity.cs:12]: warnings

File: ServiceFinder/ServiceFinder.DAL/Entites/ReviewEntity.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿using ServiceFinder.DAL.Interceptors.Interfaces;
+
+namespace ServiceFinder.DAL.Entites
+{
+    public class ReviewEntity : ISoftDeleteEntity, IAuditableEntity
+    {
+        public Guid Id { get; set; }
+        public Guid ServiceId { get; set; }
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.DAL/Entites/ReviewEntity.cs:8]: warnings

**Review by moltenessence — CHANGES_REQUESTED** [review]

**Review by moltenessence — APPROVED** [review]

**Review by pavelGagarin0870 — APPROVED** [review]

**Pull request merged**